clicommand/oidc_token.go

```diff
@@ -0,0 +1,147 @@
+package clicommand
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/buildkite/agent/v3/api"
+	"github.com/buildkite/agent/v3/cliconfig"
+	"github.com/buildkite/roko"
+	"github.com/urfave/cli"
+)
+
+type OidcTokenConfig struct {
+	Audience string `cli:"audience"`
+	Job      string `cli:"job"      validate:"required"`
+
+	// Global flags
+	Debug       bool     `cli:"debug"`
+	LogLevel    string   `cli:"log-level"`
+	NoColor     bool     `cli:"no-color"`
+	Experiments []string `cli:"experiment" normalize:"list"`
+	Profile     string   `cli:"profile"`
+
+	// API config
+	DebugHTTP        bool   `cli:"debug-http"`
+	AgentAccessToken string `cli:"agent-access-token" validate:"required"`
+	Endpoint         string `cli:"endpoint"           validate:"required"`
+	NoHTTP2          bool   `cli:"no-http2"`
+}
+
+const (
+	oidcTokenDescription = `Usage:
+
+   buildkite-agent oidc token [options...]
+
+Description:
+   Requests and prints an OIDC token from Buildkite that claims the Job ID and
+	 the specified audience. If no audience is specified, the endpoint's default
+	 audience will be claimed.
+
+Example:
+   $ buildkite-agent oidc token --audience sts.amazonaws.com
+
+   Prints the environment passed into the process
+`
+	backoffSeconds = 2
+	maxAttempts    = 5
+)
+
+var OidcTokenCommand = cli.Command{
+	Name:        "token",
```

Comment:
I wonder if something verby like `request-token` or `get-token` would be clearer than the noun `token`?

The verb “request” would align with https://openid.net/specs/openid-connect-core-1_0.html#TokenRequest

> A Client makes a Token Request by presenting its Authorization Grant (in the form of an Authorization Code) to the Token Endpoint …